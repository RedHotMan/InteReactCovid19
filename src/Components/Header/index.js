import React from "react";
import {
  StyledAppBar,
  StyledToolbar,
  StyledTypographyTitle,
  GithubBtn,
} from "./style";

export default function Header() {
  return (
    <StyledAppBar position="static">
      <StyledToolbar>
        <i className="ri-earth-line"></i>
        <StyledTypographyTitle variant="subtitle2">
          InteReactCovid19
        </StyledTypographyTitle>
        <GithubBtn
          onClick={() =>
            window.open(
              "https://github.com/RedHotMan/InteReactCovid19",
              "_self"
            )
          }
        >
          <i className="ri-github-fill"></i>
        </GithubBtn>
      </StyledToolbar>
    </StyledAppBar>
  );
}
