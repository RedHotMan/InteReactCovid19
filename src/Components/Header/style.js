import AppBar from "@material-ui/core/AppBar";
import Toolbar from "@material-ui/core/Toolbar";
import Typography from "@material-ui/core/Typography";
import IconButton from "@material-ui/core/IconButton";

import styled from "styled-components";

export const StyledAppBar = styled(AppBar)`
  background-color: #1f3349 !important;
`;

export const StyledToolabr = styled(Toolbar)`
  height: 2.5rem;
  min-height: 0 !important;
`;

export const StyledTypographyTitle = styled(Typography)`
  margin: 0 0.5rem !important;
  flex-grow: 1;
`;

export const GithubBtn = styled(IconButton)`
  &&& {
    color: #fff;
  }
`;
