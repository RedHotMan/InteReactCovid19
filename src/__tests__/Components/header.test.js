import React from "react";
import { shallow } from "enzyme";
import Header from "../../Components/Header";
import {
  StyledAppBar,
  StyledToolbar,
  StyledTypographyTitle,
  GithubBtn,
} from "../../Components/Header/style";

describe("Header", () => {
  const wrapper = shallow(<Header />);

  it("Should render Header correctly", () => {
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it("Should render a TopBar with a title and github button", () => {
    expect(wrapper.find(StyledAppBar)).toHaveLength(1);

    const appBar = wrapper.find(StyledAppBar);

    expect(appBar.find(StyledToolbar)).toHaveLength(1);

    const toolBar = appBar.find(StyledToolbar);

    expect(toolBar.contains(<i className="ri-earth-line"></i>)).toBeTruthy();
    expect(toolBar.contains("InteReactCovid19")).toBeTruthy();
    expect(toolBar.find(StyledTypographyTitle)).toHaveLength(1);
    expect(toolBar.find(GithubBtn)).toHaveLength(1);

    const gitHubBtn = toolBar.find(GithubBtn);

    expect(gitHubBtn.contains(<i className="ri-github-fill"></i>)).toBeTruthy();
  });
});
