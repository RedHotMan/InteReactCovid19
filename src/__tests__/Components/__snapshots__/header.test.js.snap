// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Should render Header correctly 1`] = `
"<Styled(WithStyles(ForwardRef(AppBar))) position=\\"static\\">
  <Styled(WithStyles(ForwardRef(Toolbar)))>
    <i className=\\"ri-earth-line\\" />
    <Styled(WithStyles(ForwardRef(Typography))) variant=\\"subtitle2\\">
      InteReactCovid19
    </Styled(WithStyles(ForwardRef(Typography)))>
    <Styled(WithStyles(ForwardRef(IconButton))) onClick={[Function: onClick]}>
      <i className=\\"ri-github-fill\\" />
    </Styled(WithStyles(ForwardRef(IconButton)))>
  </Styled(WithStyles(ForwardRef(Toolbar)))>
</Styled(WithStyles(ForwardRef(AppBar)))>"
`;
